crypto/keys/private.go

```diff
@@ -23,8 +24,16 @@ import (
 	"errors"
 	"fmt"
 	"io/ioutil"
+
+	"github.com/google/trillian"
 )
 
+// Provider handles acquisition of signers for trees.
+type Provider interface {
```

Comment:
At least a portion of the key generation will be provider-specific though, so it can't be just a utility method. PEMKeyProvider *could* implement it, but I'm cautious about doing so because of the security implications. I've got [a branch that implements it](https://github.com/RJPercival/trillian/blob/pemkey_generate/crypto/keys/pem_key_provider.go) if you'd like to see how it could be done.

I'll rename `keys.Provider` to `keys.SignerFactory`, as you make a good argument.